packages/wallets/src/features/cashier/flows/WalletWithdrawal/WalletWithdrawal.tsx

```diff
@@ -11,21 +11,31 @@ import {
 
 const WalletWithdrawal = () => {
     const { isSuccess: isCurrencyConfigSuccess } = useCurrencyConfig();
+    const { switchAccount } = useAuthorize();
     const { data: activeWallet } = useActiveWalletAccount();
     const [verificationCode, setVerificationCode] = useState('');
 
     useEffect(() => {
         const queryParams = new URLSearchParams(location.search);
+        const loginidQueryParam = queryParams.get('loginid');
         const verificationQueryParam = queryParams.get('verification');
 
+        // if loginid query param doesn't match active wallet's loginid on mount, initiate account switching
+        if (loginidQueryParam && loginidQueryParam !== activeWallet?.loginid) {
```

Comment:
since we really badly want to get rid of all switchAccount() from random commponents, 
and since the logic to set up initial account is available here in that PR to be merged soon:
https://github.com/binary-com/deriv-app/pull/14183/files

would it make sense to accomodate for this logic/business requirment within "selectDefaultAccount" account function from the PR above?

this way, it lowers the chances of getting race hazards and we can have one place deciding which account is to be loaded (instead having it spread across multiple components and thus happening very much in unpredictable manner)